Source/AwsCommonRuntimeKit/io/HostResolver.swift

```diff
@@ -30,26 +28,22 @@ public final class DefaultHostResolver: HostResolver {
         let shutdownCallbackOptionPtr = shutDownOptions?.toShutDownCPointer()
 
         self.shutDownOptions = shutDownOptions
-        let options = aws_host_resolver_default_options(max_entries: maxHosts,
-                                                        el_group: elg.rawValue,
-                                                        shutdown_options: shutdownCallbackOptionPtr,
-                                                        system_clock_override_fn: nil)
-        self.hostResolverOptionsPtr = fromPointer(ptr: options)
-        self.rawValue = aws_host_resolver_new_default(allocator.rawValue, hostResolverOptionsPtr)
-
-        let config = aws_host_resolution_config(
-            impl: aws_default_dns_resolve,
-            max_ttl: maxTTL,
-            impl_data: nil
-        )
-
-        self.config = fromPointer(ptr: config)
 
+        var options = aws_host_resolver_default_options(max_entries: maxHosts,
+                el_group: elg.rawValue,
+                shutdown_options: shutdownCallbackOptionPtr,
+                system_clock_override_fn: nil)
+        rawValue = aws_host_resolver_new_default(allocator.rawValue, &options)
+        config = try! allocator.allocate(capacity: 1)
```

Comment:
Try! is to just crash the application by throwing exception if we can’t allocate memory. I think swift also uses this approach (https://forums.swift.org/t/allocating-memory-without-a-crash/29758/3). Alternatively, we can just throw an exception which Swift SDK can handle.

Reply:
you might want to confirm we still need this. looks like allocate implementation relies on aws_mem_calloc returning null in guard let statement to throw exception, but calloc no longer returns null in latest crt and instead exits.

Reply:
I'm not sure why `config` is a member variable. This is just another "options struct" for calls to `aws_host_resolver_resolve_host()`.

 (and that function is another one that failed to mark the argument as `const`, which is why Swift is making us do weird tricks with the pointer types)